src/midi/midi_fluidsynth.cpp

```diff
@@ -0,0 +1,245 @@
+
+	auto get_double = [section](const char *const propname) {
+		try {
+			return std::stod(section->Get_string(propname));
+		} catch (const std::exception &e) {
+			/*
+			log_cb(RETRO_LOG_WARN,
+			       "[dosbox] error reading floating point '%s' "
+			       "conf setting: %s\n",
+			       e.what());
+		       */
+			return 0.0;
+		}
+	};
+
+	fluid_settings_setnum(settings.get(), "synth.sample-rate",
+	                      section->Get_int("fluid.samplerate"));
+	fluid_settings_setnum(settings.get(), "synth.gain", get_double("fluid.gain"));
+	fluid_settings_setint(settings.get(), "synth.polyphony",
+	                      section->Get_int("fluid.polyphony"));
+	fluid_settings_setint(settings.get(), "synth.cpu-cores",
+	                      section->Get_int("fluid.cores"));
+
+	fluid_settings_setint(settings.get(), "synth.reverb.active",
+	                      section->Get_bool("fluid.reverb"));
+	fluid_settings_setnum(settings.get(), "synth.reverb.room-size",
+	                      get_double("fluid.reverb.roomsize"));
+	fluid_settings_setnum(settings.get(), "synth.reverb.damp",
+	                      get_double("fluid.reverb.damping"));
+	fluid_settings_setnum(settings.get(), "synth.reverb.width",
+	                      get_double("fluid.reverb.width"));
+	fluid_settings_setnum(settings.get(), "synth.reverb.level",
+	                      get_double("fluid.reverb.level"));
+
+	fluid_settings_setint(settings.get(), "synth.chorus.active",
+	                      section->Get_bool("fluid.chorus"));
+	fluid_settings_setint(settings.get(), "synth.chorus.nr",
+	                      section->Get_int("fluid.chorus.number"));
+	fluid_settings_setnum(settings.get(), "", get_double("fluid.chorus.level"));
+	fluid_settings_setnum(settings.get(), "synth.chorus.speed",
+	                      get_double("fluid.chorus.speed"));
+	fluid_settings_setnum(settings.get(), "synth.chorus.depth",
+	                      get_double("fluid.chorus.depth"));
+
+	fsynth_ptr_t fluid_synth(new_fluid_synth(settings.get()), delete_fluid_synth);
+	if (!fluid_synth) {
+		// log_cb(RETRO_LOG_WARN, "[dosbox] Error creating fluidsynth synthesiser\n");
+		return false;
+	}
+
+	std::string soundfont = section->Get_string("fluid.soundfont");
+	if (!soundfont.empty() && fluid_synth_sfcount(fluid_synth.get()) == 0) {
+		fluid_synth_sfload(fluid_synth.get(), soundfont.data(), true);
+	}
+
+	mixer_channel_ptr_t mixer_channel(
+	        MIXER_AddChannel(mixerCallback,
+	                         section->Get_int("fluid.samplerate"), "FSYNTH"),
+	        MIXER_DelChannel);
+	mixer_channel->Enable(true);
+
+	fluid_settings = std::move(settings);
+	synth = std::move(fluid_synth);
+	channel = std::move(mixer_channel);
+	is_open = true;
+	return true;
+}
+
+void MidiHandlerFluidsynth::Close()
+{
+	if (!is_open)
+		return;
+
+	channel->Enable(false);
+	channel = nullptr;
+	synth = nullptr;
+	fluid_settings = nullptr;
```

Comment:
Loving the private type declarations, unique_pointers, and move's - outstanding @dreamer. Definitely a model for rolling out this approach into other allocating classes!